server/src/diskdb.ts

```diff
@@ -0,0 +1,85 @@
+import { promises as fs } from 'fs';
+import {
+  Result,
+  Database,
+  DatabaseType,
+  DatabaseModel,
+} from '../../frontend/src/types';
+import * as uuid from 'uuid';
+
+export class Disk implements Database {
+  public type = DatabaseType.Disk;
+  db: Map<string, DatabaseModel>;
+  path = './db.json';
+
+  async saveDB() {
+    const serialiseDB = JSON.stringify(Array.from(this.db.entries()));
+    await fs.writeFile(this.path, serialiseDB);
+  }
+
+  async fetchDB() {
+    try {
+      const file = await fs.readFile(this.path, 'utf-8');
+      this.db = new Map(JSON.parse(file));
+    } catch (err) {
+      throw `Could not open the file ${this.path} with error: ${err}`;
+    }
+  }
+
+  public async init() {
+    try {
+      await this.fetchDB();
+    } catch (err) {
+      // No database found so create a new one
+      this.db = new Map();
+    }
+  }
+
+  public async storeProject(projectInfo: Result): Promise<string> {
```

Comment:
`Result` as type for a function parameter seems a little confusing. I would expect to use `DatabaseModel` as the parameter type and push the business logic of remapping the UI data model up the stack